### DISCUSSION TITLE
Heroku no more free dyno hours

### DISCUSSION THREAD

**Discussion by thib5** [babybuddy/babybuddy#553] [Support]
Hi, I have used the entire allocation of 550 free dyno hours for the month and i read thant heroku will charge starting November 28th, 2022..

I will host the server at my house ( In my home assistant server )  but is there an easy way to keep my data ? 

Thanks

**Comment by cdubz** [comment]
You should be able to export the data from the Heroku database instance. See [Importing and Exporting Heroku Postgres Databases](https://devcenter.heroku.com/articles/heroku-postgres-import-export) for some details.

**Comment by thib5** [comment]
Thank you i will check later ! after how can I import it back in my new server in home assistant ?

**Reply by cdubz** [reply]
I'm not familiar with HA so unsure on that. @jcgoette any chance you have some advice here?

**Reply by jcgoette** [reply]
I'm wondering if @thib5 is talking about self-hosting BB within HA with https://github.com/OttPeterR/addon-babybuddy? Or if they plan on self-hosting outside of HA?

**Reply by thib5** [reply]
I will host it in HA like this : https://github.com/OttPeterR/addon-babybuddy

**Reply by jcgoette** [reply]
Calling @OttPeterR.

**Reply by OttPeterR** [reply]
I'm not aware of any data import feature in BB so you might have to get your hands dirty a little. This is new territory for me, so I'm guessing right now. 
1. Do an export that is compatible with sqlite (since BB uses sqlite in HA) [this might help](https://stackoverflow.com/questions/6148421/how-to-convert-a-postgres-database-to-sqlite)
1. Install BB on HA and launch it, so it creates the database and everything. Keep it running
1. Transfer your database export to HA (via sftp, samba, or whatever you'd like)
1. Then transfer that database export from HA to BB via `docker cp` (find the BB container with `docker ps`)
1. (uncertain) import your file into the existing BB sqlite db with the [`.import`](https://sqlite.org/cli.html) function. I think the database is located here: ` /data/data/db.sqlite3` but can @cdubz confirm this is the right thing?

**Reply by cdubz** [reply]
Sounds about right but also pretty complicated 😞

Baby Buddy does actually have a built-in per-model export/import feature. I’ve never taken the time to get it fully integrated so it’s a bit rough around the edges but probably easier than the Postgres->SQLite conversion.

Documentation is here: http://docs.baby-buddy.net/import-export/. In this case you could set up a new instance and then export the data for each model and import it to the new instance.
